vcr/issuer/openid4vci/issuer.go

```diff
@@ -124,40 +124,46 @@ func (i *issuer) ProviderMetadata(issuer did.DID) (oidc4vci.ProviderMetadata, er
 	}, nil
 }
 
-func (i *issuer) HandleAccessTokenRequest(ctx context.Context, issuer did.DID, preAuthorizedCode string) (string, error) {
+func (i *issuer) HandleAccessTokenRequest(ctx context.Context, issuer did.DID, preAuthorizedCode string) (string, string, error) {
 	flow, err := i.store.FindByReference(ctx, preAuthCodeRefType, preAuthorizedCode)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 	if flow == nil {
-		return "", oidc4vci.Error{
+		return "", "", oidc4vci.Error{
 			Err:        errors.New("unknown pre-authorized code"),
 			Code:       oidc4vci.InvalidGrant,
 			StatusCode: http.StatusBadRequest,
 		}
 	}
 	if flow.IssuerID != issuer.String() {
-		return "", oidc4vci.Error{
+		return "", "", oidc4vci.Error{
 			Err:        errors.New("pre-authorized code not issued by this issuer"),
 			Code:       oidc4vci.InvalidGrant,
 			StatusCode: http.StatusBadRequest,
 		}
 	}
 	accessToken := generateCode()
-	err = i.store.StoreReference(ctx, flow.ID, accessTokenRefType, accessToken, time.Now().Add(ttl))
+	err = i.store.StoreReference(ctx, flow.ID, accessTokenRefType, accessToken, time.Now().Add(TokenTTL))
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
+	cNonce := generateCode()
+	err = i.store.StoreReference(ctx, flow.ID, cNonceRefType, cNonce, time.Now().Add(TokenTTL))
```

Comment:
could be a second clock drift between these two TTLs due to rounding? probably doesn't matter

Reply:
same server... takes days to generate a drift of seconds.